<?xml version="1.0"?>
<launch>

<!-- launch file that starts the micp_eval node. For experiments only -->
<!-- for production, use micp_localization node instead -->
<arg name="map" default="/put/your/mesh/map/path/here.ply" description="path to map file" />
<arg name="generate_evaluation" default="false" description="generates evaluation files for accuracies. Slows down the localization." />
<arg name="initial_guess" default="[0.0, 0.0, 0.0, 0.0, 0.0, 0.0]" description="initial guess. either x,y,z,r,p,y or x,y,z,qx,qy,qz,qw" />

<node pkg="rmcl" exec="micp_eval" name="micp_eval" output="screen">
    <param name="map_file" value="$(var map)" />
    <param name="initial_guess" value="$(var initial_guess)" />
    
    <param name="lidar_frame" value="velodyne"/>
    <param name="base_frame" value="base_footprint" />
    <param name="odom_frame" value="odom" />
    <param name="map_frame" value="map" />

    <remap from="ouster/points" to="velodyne/points" />

    <param name="disable_registration" value="false" />
    <param name="generate_evaluation" value="$(var generate_evaluation)" />

    <param name="correction_mode" value="1" />
    <param name="iterations" value="20" />
    <param name="max_distance" value="3.0" />
</node>

</launch>